### PR TITLE
Add chat to demo

### PR DESCRIPTION
**Pull request by nienkedekker**
This PR adds a simple chat interface to the demo application:

Screenshot 2022-06-10 at 10 49 29

A button to toggle the chat on and off has been added to Tray. The icon in this button will be highlighted when there's a new chat message and the local participant does not have chat open.

Demo can be found here: https://custom-video-daily-react-hooks-v3.netlify.app/

### CODE REVIEW

**Review by jayne-mast — APPROVED** [review]

File: src/components/Chat/Chat.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          },
+        ]),
+      [],
+    ),
+  });
+
+  const sendMessage = useCallback(
+    (message) => {
+      /* Send the message to all participants in the chat - this does not include ourselves!
+       * See https://docs.daily.co/reference/daily-js/events/participant-events#app-message
```
Comment (jessmitch42) [inline src/components/Chat/Chat.js:29]: nice 😎

**Review by jessmitch42 — APPROVED** [review]
Lgtm!!
Two pieces of optional feedback:
- I think it's nice to have an X close button on the top right of the chat component because it's not always super obvious you can click the tray button to toggle. 
- Currently, you can send empty strings as a message. We might want to add an earlier return in this case so there's an actual message to send. 
I'll leave both of those to you to decide though!

**Review by Regaddi — APPROVED** [review]
Usage of Daily React Hooks looks **completely** fine to me 😎

**Comment by nienkedekker** [comment]
> Lgtm!! Two pieces of optional feedback:
> 
> * I think it's nice to have an X close button on the top right of the chat component because it's not always super obvious you can click the tray button to toggle.
> * Currently, you can send empty strings as a message. We might want to add an early return in this case so there's an actual message to send.
>   I'll leave both of those to you to decide though!

Check and check! 🙏

**Pull request merged**